src/uid2CookieManager.ts

```diff
@@ -71,12 +71,20 @@ export class UID2CookieManager {
     }
   }
 
+  private migrateLegacyCookie(identity: LegacyUid2SDKCookie, now: number): Uid2Identity {
+    const newCookie = enrichIdentity(identity, now);
+    this.setCookie(newCookie);
```

Comment:
If the publisher is close to the cookie size limit, there's a chance this will fail - I think it'd be better to stick with the existing approach of enriching it and returning it, rather than storing the enriched cookie.

Reply:
The enriched cookie is just the new cookie right? How do we plan to migrate from legacy cookie to the new (enriched) cookie if not by storing the enriched cookie? Can we just have them use local storage if they are close to the cookie size limit?

Reply:
There's a good chance we'd be going from the legacy cookie straight into local storage, so if we just return it enriched here and store it in whatever is trying to upgrade it, then the common case won't ever increase the cookie size.

Reply:
Actually - my comment's not quite right.

A legacy cookie will eventually get replaced by a new one - the next time the identity is updated (probably when it gets refreshed). So if we just enrich it when we read it (and don't write it back), it'll eventually be stored in the new format - hopefully in local storage instead. But that then breaks the load, because the migrated value will look newer than anything in local storage. Hmmm.

Reply:
Thinking through a few options, I think maybe your initial approach is the right move here - just upgrade it and store it back immediately.